### DISCUSSION TITLE
`onupdate` functionality not available?

### DISCUSSION THREAD

**Discussion by Julian-J-S** [fastapi/sqlmodel#990] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlalchemy import func, DateTime

# Working with FastAPI / sqlalchemy

created_at = mapped_column(DateTime, server_default=func.now())
updated_at = mapped_column(DateTime, onupdate=func.now())

created_at = Column(DateTime, server_default=func.now())
updated_at = Column(DateTime, onupdate=func.now())


# SQLModel

created_at: datetime | None = Field(default=func.now())
updated_at = ??????????????
```

### Description

I would like my models / tables to have a `created_at` and `updated_at` field that automatically get set on creation or update.

When working with FastAPI / sqlalchemy I have multple options to specify that using
- `server_default`: which creates the timestamp on the db
- onupdate: set value on each update on the db

As far as I have seen in the docs there is only a
- `default`: probably also called by the db itself

but there is no `onupdate` or similar?

I tried to integrete the sqlalchemy fields into SQLModel but this did not work.
Any way to achieve this or is this something thats in development? 
thanks!

### Operating System

Windows

### SQLModel Version

0.0.19

### Python Version

3.10

**Comment by redb0** [comment, marked as answer]
You can use `sa_column_kwargs`. My example of using as an mixin class

```python
class UpdatedAt(SQLModel):
    updated_at: Optional[datetime] = Field(
        default=None,
        sa_type=sa.DateTime(timezone=True),
        sa_column_kwargs={
            'onupdate': sa.func.now(),
            'server_default': sa.func.now(),
        },
    )
```

**Comment by Blackwidow-sudo** [comment]
Would be nice to have native support in `Field()` tho.
I did it with:
```py
created_at: datetime = Field(sa_column=Column(DateTime, default=func.now()))
updated_at: datetime = Field(sa_column=Column(DateTime, default=func.now(), onupdate=func.now()))
```

**Comment by KunxiSun** [comment]
I set the updater id and update time in the add function👇🏻

```python
from datetime import datetime

from sqlmodel import Session as _Session

class Session(_Session):

    def add(self, instance: BaseTable, *args, **kwargs) -> None:

        user_id = self.info['user_id']

        if isinstance(instance, CreatorUpdaterMixin) and user_id:
     
            if not instance.creator:
                instance.created_by = user_id
                instance.create_at = datetime.now()
            else:
                instance.updated_by = user_id
                instance.update_at = datetime.now()
```
